src/components/input/input.js

```diff
@@ -480,6 +530,7 @@ function placeholderDirective($log) {
   }
 }
 
+var visibilityDirectives = ['ngIf', 'ngShow', 'ngHide', 'ngSwitchWhen', 'ngSwitchDefault'];
```

Comment:
I added the following to the docs:

> We automatically disable this auto-hiding functionality if you provide any of the following visibility directives on the `ng-messages` container:
 - `ng-if`
 - `ng-show`/`ng-hide`
 - `ng-switch-when`/`ng-switch-default`
>
> You can also disable this functionality manually by adding the `md-auto-hide="false"` expression to the `ng-messages` container. This may be helpful if you always want to see the error messages or if you are building your own visibilty directive.

Is this sufficient, or should I add something else?